### PR TITLE
Add backwards compatibility for passing row values in console functions

### PR DESCRIPTION
**Pull request by kpurcell84**
Followup on: #2067

I realized that after the changes in #2067, you can no longer use many of the Console functions in the convenient ways you used to, so I added some backwards compatibility to the deserializer which should help to break old Console scripts less, and provide the user with more flexibility.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2080)

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -165,7 +165,7 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/2048>`__)
 
     *    - |improved|
-         - Improved the way rows and named columns are outputted in AtlasConsole to be more intuitive and easier to use. Note that this may break existing AtlasConsole scripts.
+         - Improved the way rows and named columns are output in AtlasConsole to be more intuitive and easier to use. Note that this may break existing AtlasConsole scripts.
```
Comment (gbonik) [inline docs/source/release_notes/release-notes.rst]: You probably don't want to change a release note from an older PR

File: atlasdb-service/src/main/java/com/palantir/atlasdb/jackson/AtlasDeserializers.java [hunk 2]
```diff
@@ -78,17 +79,25 @@ private AtlasDeserializers() {
...
-            bytes[i] = component.getType().convertFromJson(value.toString());
+            bytes[i] = component.getType().convertFromJson(rowValues.next().toString());
             if (component.isReverseOrder()) {
                 EncodingUtils.flipAllBitsInPlace(bytes[i]);
             }
         }
         return Bytes.concat(bytes);
     }
 
+    private static Iterator<JsonNode> getComponentNodes(JsonNode node, List<NameComponentDescription> components) {
```
Comment (gbonik) [inline atlasdb-service/src/main/java/com/palantir/atlasdb/jackson/AtlasDeserializers.java]: We need a test for this

File: atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy [hunk 3]
```diff
@@ -50,35 +50,62 @@ class AtlasCoreModule implements AtlasConsoleModule {
                          Ex: def x = table('stream_value')
 
                          To retrieve metadata about a table, use <TABLE>.describe(),
-                         <TABLE>.isDynamic(), <TABLE>.rowComponents(), and <TABLE>.columnNames().'''.stripIndent(),
+                         <TABLE>.isDynamic(), <TABLE>.rowComponents(), and <TABLE>.columnNames().
+                         '''.stripIndent(),
+
             'transactions': '''\
                          Atomic transactions used to avoid read/write or write/write
```
Comment (gbonik) [inline atlasdb-console/src/main/groovy/com/palantir/atlasdb/console/module/AtlasCoreModule.groovy]: This statement doesn't make any sense. Seems like the person who wrote it didn't understand what a transaction is.

**Comment by kpurcell84** [comment]
Note: this is not ready for merge yet. There's more testing I want to do to see if I broke any other functions with the output changes in #2067.

**Comment by kpurcell84** [comment]
Okay @gbonik, @gmaretic, @hsaraogi I've manually tested these fixes pretty thoroughly and added some unit tests for the serializers/deserializers. Ready to merge if it looks good (and circle tests pass).

**Comment by fsamuel-bs** [comment]
I've rebased on develop to fix the test infra. Merging it now since we want to release 0.46 today, and this change should be in the same version as #2067.

**Pull request merged**